### PR TITLE
COMMERCE-9477 NullPointerException after using PATCH API: /v1.0/orders/{id} to adjust an order

### PR DESCRIPTION
**Pull request by ghost**
https://issues.liferay.com/browse/COMMERCE-9477

https://github.com/CrescenzoRegaLF/liferay-portal/pull/6
https://github.com/liferay-commerce/liferay-portal/pull/2785

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java [hunk 1]
```diff
@@ -952,15 +976,19 @@ public CommerceOrderItem updateCommerceOrderItemPrices(
 
 		validateParentCommerceOrderId(commerceOrderItem);
 
-		commerceOrderItem.setDiscountAmount(discountAmount);
-		commerceOrderItem.setDiscountPercentageLevel1(discountPercentageLevel1);
-		commerceOrderItem.setDiscountPercentageLevel2(discountPercentageLevel2);
-		commerceOrderItem.setDiscountPercentageLevel3(discountPercentageLevel3);
-		commerceOrderItem.setDiscountPercentageLevel4(discountPercentageLevel4);
-		commerceOrderItem.setFinalPrice(finalPrice);
+		commerceOrderItem.setDiscountAmount(_getNonnullValue(discountAmount));
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java:979]: I believe you can use GetterUtil here without creating a new method

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java [hunk 2]
```diff
@@ -522,10 +528,19 @@ private CommerceOrder _updateNestedResources(
...
+			String[] strings = _getOrderItemExternalReferenceCodes(orderItems);
+
+			if (ArrayUtil.isEmpty(longs)) {
+				longs = new Long[] {0L};
+			}
+
+			if (ArrayUtil.isEmpty(strings)) {
+				strings = null;
+			}
+
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java:541]: - I think this is not needed the empty cases are already managed with the change made in the local service
- I'm not sure assign null to strings won't cause any trouble to the service
- I'm not sure put 0 as default value for id is right

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java [hunk 3]
```diff
@@ -522,10 +528,19 @@ private CommerceOrder _updateNestedResources(
...
 			_commerceOrderItemService.deleteMissingCommerceOrderItems(
-				commerceOrder.getCommerceOrderId(),
-				_getOrderItemIds(orderItems),
-				_getOrderItemExternalReferenceCodes(orderItems));
+				commerceOrder.getCommerceOrderId(), longs, strings);
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java:543]: I think you can leave the code as it was before here

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:stop

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Review by andrea-ale-sbarra — APPROVED** [review]
I've worked along with @CrescenzoRegaLF on this and I saw this code running

**Comment by pietrobocale** [comment]
QA PASSED

**Comment by pietrobocale** [comment]
ci:forward

**Comment by pietrobocale** [comment]
ci:forward

**Comment by brittneyq** [comment]
ci:forward

**Pull request closed without merging**